src/main/java/io/cryostat/discovery/DiscoveryStorage.java

```diff
@@ -143,34 +147,53 @@ public void start(Promise<Void> future) throws Exception {
                                                                                 + " deployed")))
                 .onFailure(future::fail);
 
-        this.timerId = getVertx().setPeriodic(pingPeriod.toMillis(), i -> pingPrune());
+        this.pluginPruneTimerId = getVertx().setPeriodic(pingPeriod.toMillis(), i -> pingPrune());
+        this.targetRetryTimeId =
+                getVertx()
+                        .setPeriodic(
+                                // TODO make this configurable, use an exponential backoff, have a
+                                // maximum retry policy, etc.
```

Comment:
Are we planning these configurations for this PR?

Reply:
I thought about it, but I'm not sure it's worth investing the time/effort into it. This is a narrow enough corner case that a single retry after 15 seconds is almost certain to succeed, so the exponential backoff wouldn't do anything. Maximum retries could be useful but it would be something that the end user needs to decide to configure, which I doubt would happen. Similarly for changing the periodicity, I'm not sure anyone would actually bother tuning the parameter. If this case is run into very frequently then the better solution is probably not to tune this parameter but instead to switch to using the Agent.

Reply:
Ahh right! Sounds good! Thanks!